tests/validation/tests/v3_api/test_websocket.py

```diff
@@ -0,0 +1,154 @@
+from threading import Thread
+from .common import CATTLE_TEST_URL
+from .common import create_kubeconfig
+from .common import get_user_client_and_cluster
+from .common import get_project_client_for_token
+from .common import USER_TOKEN
+import websocket
+import ssl
+import base64
+import time
+import pytest
+
+last_message = ""
+namespace = {"cluster": None, "shell_url": None, "pod": None, "ns": ""}
+
+
+def send_a_command(ws_connection, command):
+    cmd_enc = base64.b64encode(command.encode('utf-8')).decode('utf-8')
+    ws_connection.send('0' + cmd_enc)
+    # sends the command to the webSocket
+    ws_connection.send('0DQ==')
+    time.sleep(3)
+
+
+def test_websocket_launch_kubectl():
+    global last_message
+    ws = websocket.create_connection(
```

Comment:
move connection and receiver daemon spawn to a common function - we use it multiple times w/ similar params

Reply:
I keep this function invocation but wrap the thread creation in a new function.

Reply:
I keep this function invocation but wrap the thread creation in a new function.

Reply:
There's still a lot of duplicate code because we're passing the same options (except for url) in every test.

Reply:
This is sovled by adding a new function